uvm/plan9.go

```diff
@@ -0,0 +1,129 @@
+package uvm
+
+import (
+	"fmt"
+	//"path/filepath"
+	"strings"
+
+	"github.com/Microsoft/hcsshim/internal/schema2"
+	"github.com/Microsoft/hcsshim/internal/wclayer"
+	"github.com/Microsoft/hcsshim/uvm/lcowhostedsettings"
+	"github.com/sirupsen/logrus"
+)
+
+// AddPlan9 adds a Plan9 share to a utility VM. Each Plan9 share is ref-counted and
+// only added if it isn't already.
+func (uvm *UtilityVM) AddPlan9(hostPath string, uvmPath string, flags int32) error {
+	if uvm.operatingSystem != "linux" {
+		return errNotSupported
+	}
+	if uvmPath == "" {
+		return fmt.Errorf("uvmPath must be passed to AddPlan9")
+	}
+
+	hostPath = strings.ToLower(hostPath)
+	logrus.Debugf("uvm::AddPlan9 %s %s %d id:%s", hostPath, uvmPath, flags, uvm.id)
+	uvm.m.Lock()
+	defer uvm.m.Unlock()
+	if uvm.plan9Shares == nil {
+		uvm.plan9Shares = make(map[string]plan9Info)
+	}
+	if _, ok := uvm.plan9Shares[hostPath]; !ok {
+		guid, err := wclayer.NameToGuid(hostPath) // TODO: We should hash the full hostpath on VSMB too
```

Comment:
NameToGuid is an infuriating hack. Let's not use it anywhere new. Can you just use the counter to generate a unique name?